File: src/baseDatos/Deserializador.java
package baseDatos;

import gestorAplicacion.transacciones.Bolsillo;
import gestorAplicacion.transacciones.Prestamo;
import gestorAplicacion.usuario.Cliente;
import gestorAplicacion.usuario.Cuenta;
import gestorAplicacion.usuario.CuentaAhorro;

import java.io.*;
import java.util.ArrayList;
import java.util.List;

public class Deserializador {
    private static File rutaTemp = new File("src\\baseDatos\\temp");

    public static void deserializar(Cliente cliente){
        File[] docs = rutaTemp.listFiles();
        FileInputStream fis;
        ObjectInputStream ois;

        for (File file : docs) {
            if (file.getAbsolutePath().contains("prestamos.txt")){
                try {
                    fis = new FileInputStream(file);
                    ois = new ObjectInputStream(fis);

                    Cliente.setListaCuentas((ArrayList<Cuenta>) ois.readObject());
                }catch (FileNotFoundException e){
                    e.printStackTrace();
                } catch (IOException e){
                    e.printStackTrace();
                } catch (ClassNotFoundException e){
                    e.printStackTrace();
                }
            } else if (file.getAbsolutePath().contains("bolsillos.txt")) {
                try {
                    fis = new FileInputStream(file);
                    ois = new ObjectInputStream(fis);

                    Cliente.listarCuentas().setMisBolsillos((ArrayList<Bolsillo>) ois.readObject());
                } catch (FileNotFoundException e){
                    e.printStackTrace();
                } catch (IOException e){
                    e.printStackTrace();
                } catch (ClassNotFoundException e){
                    e.printStackTrace();
                }

            } /* else if (file.getAbsolutePath().contains("pagos.txt")) {
                try {
                    fis = new FileInputStream(file);
                    ois = new ObjectInputStream(fis);

                    cliente.getCuenta().setMisBolsillos((ArrayList<Bolsillo>) ois.readObject());
                } catch (FileNotFoundException e){
                    e.printStackTrace();
                } catch (IOException e){
                    e.printStackTrace();
                } catch (ClassNotFoundException e){
                    e.printStackTrace();
                }
            } else if (file.getAbsolutePath().contains("trasnferencias.txt")) {
                try {
                    fis = new FileInputStream(file);
                    ois = new ObjectInputStream(fis);

                    cliente.getCuenta().setMisBolsillos((ArrayList<Bolsillo>) ois.readObject());
                } catch (FileNotFoundException e){
                    e.printStackTrace();
                } catch (IOException e){
                    e.printStackTrace();
                } catch (ClassNotFoundException e){
                    e.printStackTrace();
                }
            }*/
        }
    }

}

File: Python/baseDatos/Deserializador.py
import pickle
from gestorAplicacion.usuarios.cliente import Cliente
from gestorAplicacion.transacciones.bolsillo import Bolsillo
from gestorAplicacion.transacciones.pago import Pago
from gestorAplicacion.transacciones.transferencia import Transferencia

class Deserializador:
    @classmethod
    def deserializar(cls, tipo):
        if tipo == "Prestamos":
            pickfile = open("baseDatos/temp/prestamos.txt", "rb")
            Cliente.listaCuentas = pickle.load(pickfile)
            pickfile.close

        elif tipo == "Bolsillos":
            pickfile = open("baseDatos/temp/bolsillos.txt", "rb")
            Cliente.listarCuentas().misBolsillos = pickle.load(pickfile)
            pickfile.close

        elif tipo=="Pagos":
            pickfile = open("baseDatos/temp/pagos.txt", "rb")
            Pago.getPagos = pickle.load(pickfile)
            pickfile.close

        elif tipo=="Transferencias":
            pickfile = open("baseDatos/temp/transferencias.txt", "rb")
            Transferencia.lista = pickle.load(pickfile)
            pickfile.close

    def DeserializarTodo():
        Deserializador.deserializar("Prestamos")
        Deserializador.deserializar("Bolsillos")
        Deserializador.deserializar("Pagos")
        Deserializador.deserializar("Transferencias")

